src/plugins/embeddable/public/lib/containers/container.ts

```diff
@@ -53,6 +53,16 @@ export abstract class Container<
       });
   }
 
+  public setChildLoaded(embeddable: IEmbeddable) {
+    this.children[embeddable.id] = embeddable;
```

Comment:
It seems that we need a check here that embeddable wasn't removed in a meantime. 
Similar like it is done inside on `onPanelAdded` just before `embeddableLoaded` was updated

Reply:
I've implemented this check inside the `setChildLoaded` method.

There's nothing wrong with performing another check here, but I do think that this may be a leftover from when the entire dashboard state was serialized into the URL.

This is because when an embeddable hasn't loaded yet, there is no way to remove it from the dashboard via the UI. When the entire dashboard state was in the URL, pressing the browser back button would remove the most recently added panel regardless of whether it was done loading or not, which I believe is the reason this check was added in the first place.